src/UI/Component/Triggerer.php

```diff
@@ -0,0 +1,33 @@
+<?php
+namespace ILIAS\UI\Component;
+
+/**
+ * Interface Triggerer
+ *
+ * Describes a component that can trigger signals of other components on given events, such as click or hover.
+ * All supported events are abstracted with interfaces (see Clickable, Hoverable).
+ * Example: A button can trigger the show signal of a modal on click (which will open the modal on button click).
```

Comment:
Nice, I really like this verbal example in the description. It helps tremendously to quickly grasp the concept.